src/routes/intro/IntroScreen.tsx

```diff
@@ -37,21 +38,21 @@ interface IntroScreenProps {
 const IntroScreen = ({ navigation }: IntroScreenProps): JSX.Element => {
   const { data: disciplines, error, loading, refresh } = useLoadDisciplines(null)
   const { data: selectedProfessions, refresh: refreshSelectedProfessions } = useReadSelectedProfessions()
+  const isProfessionSelected = selectedProfessions && selectedProfessions.length > 0
 
-  useFocusEffect(
-    React.useCallback(() => {
-      refreshSelectedProfessions()
-    }, [refreshSelectedProfessions])
-  )
+  useFocusEffect(refreshSelectedProfessions)
 
   const navigateToDiscipline = (item: Discipline): void => {
     navigation.navigate('ProfessionSelection', {
       discipline: item
     })
   }
 
-  const navigateToHomeScreen = (): void => {
-    navigation.navigate('Home')
+  const navigateToHomeScreen = async () => {
+    if (!isProfessionSelected) {
```

Comment:
I'd just check `selectedProfessions === null`, no need to set the selected professions to `[]` if it is already `[]`

Reply:
you are right. done